Simulation Code/Base/Base.ino

```diff
@@ -1,112 +1,138 @@
 #include <Arduino.h>
 
-// Define the pins for the water level sensors
+// Define the pins for the water level sensors and other components
+//
 const int WaterLevelA = 4;
 const int WaterLevelB = 5;
 const int WaterLevelC = 6;
 const int WaterLevelD = 7;
-
-// Define the pins for the fans
 const int FanA = 9;
 const int FanB = 10;
-int FanPWM; // Variable to control the speed of the fans
-
-// Define the variables for the water pump
-unsigned long previousTime = 0; // Variable to store the last time the water level was checked
-unsigned long previousPumpTime = 0; // Variable to store the last time the pump status was checked
-const unsigned long stabilityTime = 1000; // Time in milliseconds to wait before changing the pump speed
-bool pumpStatus = false; // Variable to store the current status of the pump (true = on, false = off)
-const unsigned long pumpTimeOff = 1000; // Time in milliseconds for the pump to stay off
-const unsigned long pumpTimeOn = 5000; // Time in milliseconds for the pump to stay on
-const int WaterPump = 11; // Pin for the water pump
-int PumpPWM; // Variable to control the speed of the water pump
-
-void setup() 
-{
-  Serial.begin(9600); // Start the serial communication with the computer
-  // Set the mode of the pins
-  pinMode(WaterLevelA, INPUT);
-  pinMode(WaterLevelB, INPUT);
-  pinMode(WaterLevelC, INPUT);
-  pinMode(WaterLevelD, OUTPUT);
-  pinMode(FanA, OUTPUT);
-  pinMode(FanB, OUTPUT);
-  pinMode(WaterPump, OUTPUT);
-}
+const int WaterPump = 11;
 
-// Function to check the current water level
-void checkWaterLevel(int &waterLevelA, int &waterLevelB, int &waterLevelC) 
-{
-  // Read the value of the water level sensors
-  waterLevelA = digitalRead(WaterLevelA);
-  waterLevelB = digitalRead(WaterLevelB);
-  waterLevelC = digitalRead(WaterLevelC);
-}
+// Define a structure for water level sensors
+struct WaterLevelSensors {
+    int pinA;
+    int pinB;
+    int pinC;
+
+    void begin() {
+        pinMode(pinA, INPUT);
+        pinMode(pinB, INPUT);
+        pinMode(pinC, INPUT);
+    }
+
+    void readLevels(int &levelA, int &levelB, int &levelC) {
+        levelA = digitalRead(pinA);
+        levelB = digitalRead(pinB);
+        levelC = digitalRead(pinC);
+    }
+};
+
+// Define a structure for fan control
+struct Fans {
```

Comment:
Like the comment mentioned above. It is best to separate functions from structures. If you want to add a fan function, consider creating its own function defined elsewhere. The fan structure should only contain the initialized variables relevant for fan use.